enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,375 @@
+status: provisional
+
+
+  
+---
+
+# Share Secrets And ConfigMaps Across Namespaces via a CSI Driver
+
+
+## Release Signoff Checklist
+
+- [/] Enhancement is `implementable`
+- [/] Design details are appropriately documented from clear requirements
+- [/] Test plan is defined
+- [/] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+> 1. Is a CSI `SecretProviderClass` based implementation recommended?  Received some varying guidance.  Also, per PR
+>review from @jsafrane, of which I can only concur based on my examination, the example at https://github.com/kubernetes-sigs/secrets-store-csi-driver
+>does not appear to handle refreshing of the data after initial provisioning.  So it does not appear to be something we
+>can use as is.  But do we want to take on adding such a feature upstream to that repo? **This question most
+>likely can be deferred until implementation.**
+> 2. What will be the precise means of interacting with the host filesystem.  The hostpath plugin actually uses the 
+>simple golang `os.MkdirAll` for example when creating mount access based volumes.  Can we do that in all our supported
+>scenarios?  We've also received suggestions around tmpfs or read only bind-mounts of a directory from the CSI driver.
+>**This question most likely can be deferred until implementation.**
+> 3. SELinux implications will need to be properly understood.  Minimally, a consuming Pod needs the correct SELinux 
+>context and fs permissions with bind-mounts for example.
+>**This question most likely can be deferred until implementation.**
+> 4. The EP document in https://github.com/openshift/enhancements/pull/169 needs to be reconciled with this document in
+>some form or fashion.
+
+## Summary
+
+This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based
+means for OpenShift cluster administrators to share `Secrets` / `ConfigMaps` defined in one namespace with
+other namesapces. 
+
+Any owners/authors/users of Pods in those namespaces can opt into the consumption of the content of those shared `Secrets` / 
+ `ConfigMaps` during their Pod's execution.
+ 
+The precise means of opting in consuming the shared content are outside the scope of this proposal.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** if/when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.  Or if that PR is replaced by another, this proposal will point to that.
+
+
+## Motivation
+
+Simplify consumption of `Secrets` and `ConfigMaps` defined in one namespace in additional namespaces.
```

Comment:
```suggestion
Simplify consumption of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces.
```

Reply:
The motivation is probably more like "provide the foundational pieces for building a global/shared secret/configmap feature such that user pods in one namespace can consume secrets/configmaps from another namespace, such as provided by a cluster-admin"

Reply:
(where the rest of said feature, dependent on this work,  is designed/defined/described in the other EP)

Reply:
ok I tried to capture that in an update I'm about to push